modules/qna/src/views/full/Components/TextareaList.tsx

```diff
@@ -27,59 +26,50 @@ interface Props {
 
 const TextAreaList: FC<Props> = props => {
   const [showPicker, setShowPicker] = useState(false)
-  const {
-    items,
-    duplicateMsg,
-    updateItems,
-    keyPrefix,
-    canAddContent,
-    addItemLabel,
-    label,
-    refItems,
-    placeholder
-  } = props
+  const [localItems, setLocalItems] = useState(props.items)
+  // Generating unique keys so we don't need to rerender all the list as soon as we add or delete one element
+  const [keys, setKeys] = useState(localItems.map(x => _uniqueId(keyPrefix)))
+  const { duplicateMsg, updateItems, keyPrefix, canAddContent, addItemLabel, label, refItems, placeholder } = props
```

Comment:
I'm just suggesting thing here, feel free to do implement it or not.

What do you think of passing validation functions as props, this way custom validation can be applied locally and still be customizable. In some cases we might allow empty strings and in some other not.
```
itemValidator: (item) => boolean
itemListValidator: (itemList) => boolean
```